libs/device/models/src/lib/api-config/api-config.test.ts

```diff
@@ -0,0 +1,47 @@
+/**
+ * Copyright (c) Mudita sp. z o.o. All rights reserved.
+ * For licensing, see https://github.com/mudita/mudita-center/blob/master/LICENSE.md
+ */
+
+import { ApiConfig, ApiConfigValidator } from "./api-config"
+
+const minimumApiConfig: ApiConfig = {
+  apiVersion: "1.0.0",
+  features: ["mc-overview"],
+  productId: "productId",
+  vendorId: "vendorId",
+}
+
+describe("APIConfigValidator", () => {
+  it("should return success when correct config is validated", () => {
+    const apiConfig: ApiConfig = minimumApiConfig
+    const result = ApiConfigValidator.safeParse(apiConfig)
+    expect(result.success).toBeTruthy()
+  })
+  it("should return fail when object is empty", () => {
+    const apiConfig = {}
+    const result = ApiConfigValidator.safeParse(apiConfig)
+    expect(result.success).toBeFalsy()
+  })
+  it("should return fail when lang is incorrect", () => {
+    const apiConfig = { ...minimumApiConfig, lang: "INCORRECT" }
+
+    const result = ApiConfigValidator.safeParse(apiConfig)
+    expect(result.success).toBeFalsy()
+  })
+  it("should return fail when features array is empty", () => {
+    const apiConfig = { ...minimumApiConfig, features: [] }
+
+    const result = ApiConfigValidator.safeParse(apiConfig)
+    expect(result.success).toBeFalsy()
+  })
+  it.each(["apiVersion", "productId", "vendorId"])(
+    "should return fail when %s is missing",
+    (fieldName) => {
+      const apiConfig: any = { ...minimumApiConfig }
+      delete apiConfig[fieldName]
```

Comment:
⚪ 
```suggestion
      const { [fieldName]: _, ...apiConfig } = minimumApiConfig;
```

The syntax [fieldName]: _ in object destructuring in JavaScript or TypeScript signifies omitting the value of the key stored in the variable fieldName and assigning that value to the variable _, which is used here as a dummy variable to disregard that particular item while retaining the rest of the object in apiConfig.

TypeScript properly handles this destructuring mechanism, allowing for flexible manipulation of object structures without losing type information for the remaining properties.